Conversions/DateToDay.js

```diff
@@ -51,19 +27,25 @@ const DateToDay = (date) => {
     return new TypeError('Date is not valid.')
   }
 
-  // In case of Jan and Feb, we consider it as previous year
+  // In case of Jan and Feb:
+  // Year: we consider it as previous year
   // e.g., 1/1/1987 here year is 1986 (-1)
+  // Month: we consider value as 13 & 14 respectively
   if (month < 3) {
-    year -= 1
+    year--
+    month += 12
   }
 
   // divide year to century and yearDigit value.
   const yearDigit = (year % 100)
   const century = Math.floor(year / 100)
+
   // Apply the algorithm shown above
-  const weekDay = Math.abs((day + Math.floor((calcMonthList[month] + 1) * 2.6) + yearDigit + (yearDigit / 4) + (century / 4) - (2 * century)) % 7)
+  const weekDay = Math.abs((day + Math.floor((month + 1) * 2.6) + yearDigit + (yearDigit / 4) + (century / 4) - (2 * century)) % 7)
```

Comment:
Which formula are you implementing here?

Reply:
https://en.wikipedia.org/wiki/Zeller%27s_congruence

Reply:
You didn't answer my question. Which formula - there are multiple formulas for different calendars on that page. You will want the one which (1) doesn't have problems with JS mod and (2) is for the Gregorian (not the Julian) calendar:
![Formula](https://wikimedia.org/api/rest_v1/media/math/render/svg/6aa6f22c7650128e4ec1bb7e0a768ec39d6913cc)

Reply:
Sorry, misunderstood it.
I have implemented this one:
<img width="475" alt="image" src="https://user-images.githubusercontent.com/31822501/194569520-58ec1e6f-af32-4ebc-8a3b-9c6367164b72.png">

Reply:
No, you did not:

```javascript
Math.floor((day + Math.floor((month + 1) * 2.6) + yearDigit + (yearDigit / 4) + (century / 4) - (2 * century)) % 7)
```

* Why is there a `floor` around it? There is no floor around the formula.
* There must be a `floor` around `K/4` and `J/4`; you **may not** apply this afterwards
* This won't work for all cases due to JavaScript's [modulo operator](https://en.wikipedia.org/wiki/Zeller's_congruence#Basic_modification)

Reply:
Updated, could you please check

Reply:
You have addressed the first two points, but how have you addressed the last point?

Reply:
I have now addressed point 3 as well, as per explanation given [here](https://en.wikipedia.org/wiki/Zeller's_congruence#Basic_modification)

> The formulas rely on the mathematician's definition of [modulo](https://en.wikipedia.org/wiki/Modulo_operation) division, which means that −2 mod 7 is equal to positive 5. Unfortunately, in the truncating way most computer languages implement the remainder function, −2 mod 7 returns a result of −2. So, to implement Zeller's congruence on a computer, the formulas should be altered slightly to ensure a positive numerator. The simplest way to do this is to replace − 2J by + 5J

<img width="469" alt="image" src="https://user-images.githubusercontent.com/31822501/194718818-18f3eca3-295d-4af6-98af-8054130b7b0a.png">